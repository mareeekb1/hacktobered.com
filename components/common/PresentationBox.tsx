import {
  Box,
  Button,
  Heading,
  Img,
  Text,
  useColorModeValue as mode,
} from "@chakra-ui/react";
import { BsArrowRightCircleFill } from "react-icons/bs";
import NextLink from "next/link";
import React from "react";

interface TTextBoxSectionProps {
  testID: string;
  title: string;
  description: string;
  right?: boolean;
}

export const TextBoxSection = (props: TTextBoxSectionProps) => {
  return (
    <Box
      data-testid={props.testID}
      flex={1}
      style={
        props.right
          ? {
              display: "flex",
              alignItems: "flex-end",
              flexDirection: "column",
              width: "100%",
            }
          : {}
      }
    >
      <Heading
        as="h1"
        size="xl"
        color={mode("blue.600", "blue.300")}
        mt="8"
        fontWeight="extrabold"
        letterSpacing="tight"
      >
        {props.title}
      </Heading>
      <Text
        w={{ lg: "50%" }}
        color={mode("gray.600", "gray.400")}
        mt="4"
        fontSize="lg"
        fontWeight="medium"
        textAlign={props.right ? "end" : "start"}
      >
        {props.description}
      </Text>
    </Box>
  );
};
export const ButtonSection = () => {
  return (
    <Box
      data-testid="buttonSection"
      mt={20}
      style={{ display: "flex", justifyContent: "center", width: "100%" }}
    >
      <NextLink href="/profile" passHref>
        <Button
          data-testid="button"
          variant="ghost"
          as="a"
          rightIcon={<BsArrowRightCircleFill fontWeight={"bold"} />}
          size="lg"
          minW="210px"
          colorScheme="blue"
          height="14"
          px="8"
          fontSize={32}
        >
          Try it!
        </Button>
      </NextLink>
    </Box>
  );
};

const PresentationBox = () => {
  const sections = {
    1: {
      title: "Get proud!",
      description:
        "Now you can share your achievements with others via Instagram stories, Whatsapp status or Twitter. Don't do it manualy.",
    },
    2: {
      title: "Why should you use it?",
      description:
        "You can easily share your PR's and show them to the world just in few steps.",
    },
    3: {
      title: "How it works?",
      description:
        "So EASY! Sign in via your GitHub and see overview or share your PR's.",
    },
  };

  return (
    <Box>
      {/* IMG section */}
      <Box
        data-testid="imgBox1"
        style={{ display: "flex", justifyContent: "center" }}
      >
        <Img
          textAlign={"center"}
          w={{ lg: "30%", md: "60%", sm: "50%" }}
          pos="relative"
          zIndex="1"
          h={{ lg: "100%" }}
          objectFit="cover"
          src="hcktbrd.png"
          alt="Hacktobered hero image"
        />
      </Box>
      <TextBoxSection
        description={sections[1].description}
        title={sections[1].title}
        testID="textBox1"
        right
      />
      <TextBoxSection
        description={sections[2].description}
        title={sections[2].title}
        testID="textBox2"
      />
      <TextBoxSection
        description={sections[3].description}
        title={sections[3].title}
        testID="textBox3"
        right
      />
      <ButtonSection />
    </Box>
  );
};

export default PresentationBox;
