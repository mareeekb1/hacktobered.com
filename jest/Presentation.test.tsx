import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import PresentationBox from "../components/common/PresentationBox";
import renderer from "react-test-renderer";

describe("Check correct rendering of components", () => {
  test("Render whole PresentationBox component", () => {
    const box = render(<PresentationBox />);
    expect(box).toBeTruthy();
  });
  test("Render IMG BOX 1", async () => {
    const box = await render(<PresentationBox />).findByTestId("imgBox1");
    expect(box).toBeTruthy();
  });
  test("Has compoennt TEXT BOX 1", async () => {
    const box = await render(<PresentationBox />).findByTestId("textBox1");
    expect(box).toBeTruthy();
  });
  test("Has compoennt TEXT BOX 2", async () => {
    const box = await render(<PresentationBox />).findByTestId("textBox2");
    expect(box).toBeTruthy();
  });
  test("Has compoennt TEXT BOX 3", async () => {
    const box = await render(<PresentationBox />).findByTestId("textBox3");
    expect(box).toBeTruthy();
  });
  test("Render BUTTON section", async () => {
    const box = await render(<PresentationBox />).findByTestId("buttonSection");
    expect(box).toBeTruthy();
  });
});
describe("Match snapshot of PresentationBox", () => {
  const component = renderer.create(<PresentationBox />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Button funcionality", () => {
  const { container } = render(<PresentationBox />);
  const buttonAsLink = screen.getByTestId("button");
  expect(buttonAsLink).toHaveAttribute("href", "/profile");
  it("Should trigger click event", async () => {
    await buttonAsLink.click();
    expect(container.firstChild).toMatchSnapshot();
  });
});
