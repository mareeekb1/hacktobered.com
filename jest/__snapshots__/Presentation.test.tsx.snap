// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="css-0"
>
  <div
    className="css-0"
    data-testid="imgBox1"
    style={
      {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <img
      alt="Hacktobered hero image"
      className="chakra-image css-14icgpu"
      src="hcktbrd.png"
    />
  </div>
  <div
    className="css-1rr4qq7"
    data-testid="textBox1"
    style={
      {
        "alignItems": "flex-end",
        "display": "flex",
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <h1
      className="chakra-heading css-18l4zzf"
    >
      Get proud!
    </h1>
    <p
      className="chakra-text css-1mvrx8s"
    >
      Now you can share your achievements with others via Instagram stories, Whatsapp status or Twitter. Don't do it manualy.
    </p>
  </div>
  <div
    className="css-1rr4qq7"
    data-testid="textBox2"
    style={{}}
  >
    <h1
      className="chakra-heading css-18l4zzf"
    >
      Why should you use it?
    </h1>
    <p
      className="chakra-text css-s6jlv9"
    >
      You can easily share your PR's and show them to the world just in few steps.
    </p>
  </div>
  <div
    className="css-1rr4qq7"
    data-testid="textBox3"
    style={
      {
        "alignItems": "flex-end",
        "display": "flex",
        "flexDirection": "column",
        "width": "100%",
      }
    }
  >
    <h1
      className="chakra-heading css-18l4zzf"
    >
      How it works?
    </h1>
    <p
      className="chakra-text css-1mvrx8s"
    >
      So EASY! Sign in via your GitHub and see overview or share your PR's.
    </p>
  </div>
  <div
    className="css-1xaekgw"
    data-testid="buttonSection"
    style={
      {
        "display": "flex",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <a
      className="chakra-button css-200v6p"
      data-testid="button"
      href="/profile"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Try it!
      <span
        className="chakra-button__icon css-1hzyiq5"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          fontWeight="bold"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;

exports[`Button funcionality Should trigger click event 1`] = `null`;
